Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift

```diff
@@ -267,8 +273,10 @@ public class PubackPacket {
     }
 }
 
+// We can't mutate this class after initialization. Swift can not verify the sendability due to direct use of c pointer,
+// so mark it unchecked Sendable
```

Comment:
This makes me think that class being non-final is intentional. I think we mark it final but the actual reason is that it contains C Pointers/Structs. Same comment for all of them.

Reply:
Sorry, commented at the wrong comment. I was talking about some comments which say that we can't mark it Sendable because it's non final.

Reply:
If you look at the classes referenced in those comments, you'll notice they don’t contain any C pointers or structs. If we mark those classes as `final`, we could declare them as `Sendable` directly instead of using `@unchecked Sendable`.

I didn’t initially mark them as final because I thought we might want to extend them in the future. For example, if IoT Core were to add support for QoS2, we might need to extend `PublishResult`. Of cause, it seems unlikely to happen at this point.

I'm fine either way (mark the classes as `final` or leave them as they are.)

Reply:
> I didn’t initially mark them as final because I thought we might want to extend them in the future. For example, if IoT Core were to add support for QoS2, we might need to extend PublishResult. Of cause, it seems unlikely to happen at this point.

That should be a non-breaking change. I believe marking a class final is a breaking change because other people might be extending it, but removing the final should be fine in the future.